From 6d7ba662e980fcc6f3056173043136063e6d68db Mon Sep 17 00:00:00 2001
From: "Dr. David Alan Gilbert" <dgilbert@redhat.com>
Date: Thu, 20 Aug 2020 15:14:18 -0400
Subject: [PATCH 2/6] machine types/numa: set numa_mem_supported on old machine
 types

RH-Author: Dr. David Alan Gilbert <dgilbert@redhat.com>
Message-id: <20200820151419.14723-2-dgilbert@redhat.com>
Patchwork-id: 98197
O-Subject: [RHEL-AV 8.3.0 qemu-kvm PATCH 1/2] machine types/numa: set numa_mem_supported on old machine types
Bugzilla: 1849707
RH-Acked-by: Eduardo Habkost <ehabkost@redhat.com>
RH-Acked-by: David Hildenbrand <david@redhat.com>
RH-Acked-by: Igor Mammedov <imammedo@redhat.com>

From: "Dr. David Alan Gilbert" <dgilbert@redhat.com>

Reenable the -numa mem= syntax for old machine types, this is making
the downstream old machines behave in the same way as the upstream old
machines changed in upstream 32a354dc6c07d7.

Power already seems to have the change.

Signed-off-by: Dr. David Alan Gilbert <dgilbert@redhat.com>
Signed-off-by: Danilo C. L. de Paula <ddepaula@redhat.com>
---
 hw/arm/virt.c     | 2 +-
 hw/i386/pc_piix.c | 1 +
 hw/i386/pc_q35.c  | 1 +
 3 files changed, 3 insertions(+), 1 deletion(-)

diff --git a/hw/arm/virt.c b/hw/arm/virt.c
index f087483a04..26a7920081 100644
--- a/hw/arm/virt.c
+++ b/hw/arm/virt.c
@@ -2755,7 +2755,6 @@ static void rhel_machine_class_init(ObjectClass *oc, void *data)
     hc->plug = virt_machine_device_plug_cb;
     hc->unplug_request = virt_machine_device_unplug_request_cb;
     hc->unplug = virt_machine_device_unplug_cb;
-    mc->numa_mem_supported = true;
     mc->nvdimm_supported = true;
     mc->auto_enable_numa_with_memhp = true;
     mc->default_ram_id = "mach-virt.ram";
@@ -2860,5 +2859,6 @@ static void rhel820_virt_options(MachineClass *mc)
     rhel830_virt_options(mc);
     compat_props_add(mc->compat_props, hw_compat_rhel_8_2,
                      hw_compat_rhel_8_2_len);
+    mc->numa_mem_supported = true;
 }
 DEFINE_RHEL_MACHINE(8, 2, 0)
diff --git a/hw/i386/pc_piix.c b/hw/i386/pc_piix.c
index 4af4497a0c..bda2d9ffc8 100644
--- a/hw/i386/pc_piix.c
+++ b/hw/i386/pc_piix.c
@@ -1009,6 +1009,7 @@ static void pc_machine_rhel7_options(MachineClass *m)
     pcmc->default_nic_model = "e1000";
     m->default_display = "std";
     m->no_parallel = 1;
+    m->numa_mem_supported = true;
     machine_class_allow_dynamic_sysbus_dev(m, TYPE_RAMFB_DEVICE);
     compat_props_add(m->compat_props, pc_rhel_compat, pc_rhel_compat_len);
     m->alias = "pc";
diff --git a/hw/i386/pc_q35.c b/hw/i386/pc_q35.c
index c709460ab7..d1e3a9b575 100644
--- a/hw/i386/pc_q35.c
+++ b/hw/i386/pc_q35.c
@@ -617,6 +617,7 @@ static void pc_q35_machine_rhel820_options(MachineClass *m)
     pc_q35_machine_rhel_options(m);
     m->desc = "RHEL-8.2.0 PC (Q35 + ICH9, 2009)";
     m->alias = NULL;
+    m->numa_mem_supported = true;
     pcmc->smbios_stream_product = "RHEL-AV";
     pcmc->smbios_stream_version = "8.2.0";
     compat_props_add(m->compat_props, hw_compat_rhel_8_2,
-- 
2.27.0

